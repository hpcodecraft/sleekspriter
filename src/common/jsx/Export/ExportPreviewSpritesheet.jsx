var ExportPreviewSpritesheet = React.createClass({
  // mixins: [PostalSubscriptionMixin, FrameCanvasMixin],
  render: function() {
    return (
      <div>spritesheet preview</div>
    )
  }
});