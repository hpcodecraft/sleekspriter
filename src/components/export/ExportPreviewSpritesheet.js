import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { SpritesheetCanvas } from "../canvases";

const ExportPreviewSpritesheet = ({
  background,
  format,
  frames,
  layers,
  size,
  pixels,
  zoom,
}) => {
  const classes = {
    preview: true,
    checkerboard: !!(format === "png" || format === "gif"),
  };
  const style = {
    width: frames.x * size.width * zoom,
    height: frames.y * size.height * zoom,
  };

  background = format === "jpeg" ? background : null;

  return (
    <div className={classnames(classes)} style={style}>
      <SpritesheetCanvas
        frames={frames}
        layers={layers}
        size={size}
        zoom={zoom}
        pixels={pixels || null}
        background={background}
      />
    </div>
  );
};

ExportPreviewSpritesheet.propTypes = {
  format: PropTypes.string.isRequired,
  frames: PropTypes.object.isRequired,
  layers: PropTypes.array.isRequired,
  size: PropTypes.object.isRequired,
  pixels: PropTypes.object.isRequired,
  zoom: PropTypes.number.isRequired,
};

export default ExportPreviewSpritesheet;
