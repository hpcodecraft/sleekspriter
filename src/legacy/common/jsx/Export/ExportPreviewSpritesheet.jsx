var ExportPreviewSpritesheet = React.createClass({
  render: function() {
    var width = this.props.file.size.width * this.props.ui.export.zoom * this.props.file.frames.x,
        height = this.props.file.size.height * this.props.ui.export.zoom * this.props.file.frames.y,
        classes = {
          preview: true,
          checkerboard: (this.props.ui.export.format === 'png' || this.props.ui.export.format === 'gif') ? true : false,
        },
        style = {
          width: width,
          height: height,
        },
        background = (this.props.ui.export.format === 'png' || this.props.ui.export.format === 'gif') ? null : '#ffffff';

    return (
      <div className={classNames(classes)} style={style}>
        <SpritesheetCanvas file={this.props.file} pixels={this.props.pixels} zoom={this.props.ui.export.zoom} background={background} />
      </div>
    )
  }
});