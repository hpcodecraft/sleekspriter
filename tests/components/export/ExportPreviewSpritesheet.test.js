import ExportPreviewSpritesheet from "components/export/ExportPreviewSpritesheet";

describe("ExportPreviewSpritesheet", () => {
  let props, jsx, wrapper;
  beforeEach(() => {
    props = {
      format: "png",
      frames: {
        x: 2,
        y: 2,
      },
      layers: [],
      size: { width: 16, height: 16 },
      pixels: {},
      zoom: 1,
    };
    jsx = <ExportPreviewSpritesheet {...props} />;
    wrapper = shallow(jsx);
  });

  it("should render correctly", () => {
    expect(render(jsx)).toMatchSnapshot();
  });

  describe("PNG preview", () => {
    it("does have a checkerboard background", () => {
      expect(wrapper.hasClass("checkerboard")).toEqual(true);
    });
  });

  describe("JPG preview", () => {
    it("does not have a checkerboard background", () => {
      props.format = "jpeg";
      wrapper = shallow(<ExportPreviewSpritesheet {...props} />);
      expect(wrapper.hasClass("checkerboard")).toEqual(false);
    });
  });

  describe("GIF preview", () => {
    it("does have a checkerboard background", () => {
      props.format = "gif";
      wrapper = shallow(<ExportPreviewSpritesheet {...props} />);
      expect(wrapper.hasClass("checkerboard")).toEqual(true);
    });
  });
});
