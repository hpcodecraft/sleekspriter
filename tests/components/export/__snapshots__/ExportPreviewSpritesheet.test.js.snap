// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportPreviewSpritesheet should render correctly 1`] = `
<div
  class="preview checkerboard"
  style="width:32px;height:32px"
>
  <canvas
    height="32"
    style="width:32px;height:32px"
    width="32"
  />
</div>
`;
